File: VB/WinSolution.Module/Store.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Xpo
Imports DevExpress.ExpressApp
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl

Namespace WinSolution.Module
	<DefaultClassOptions> _
	Public Class Store
		Inherits BaseObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Private _AssignedTo As Employee
		Public Property AssignedTo() As Employee
			Get
				Return _AssignedTo
			End Get
			Set(ByVal value As Employee)
				SetPropertyValue("AssignedTo", _AssignedTo, value)
			End Set
		End Property
		Public Overrides Sub AfterConstruction()
			MyBase.AfterConstruction()
			_AssignedTo = Session.GetObjectByKey(Of Employee)(SecuritySystem.CurrentUserId)
		End Sub
		Private _Name As String
		Public Property Name() As String
			Get
				Return _Name
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Name", _Name, value)
			End Set
		End Property
	End Class
End Namespace

File: CS/WinSolution.Module/Store.cs
using System;
using DevExpress.Xpo;
using DevExpress.ExpressApp;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;

namespace WinSolution.Module {
    [DefaultClassOptions]
    public class Store : BaseObject {
        public Store(Session session) : base(session) { }
        private Employee _AssignedTo;
        public Employee AssignedTo {
            get { return _AssignedTo; }
            set { SetPropertyValue("AssignedTo", ref _AssignedTo, value); }
        }
        public override void AfterConstruction() {
            base.AfterConstruction();
            _AssignedTo = Session.GetObjectByKey<Employee>(SecuritySystem.CurrentUserId);
        }
        private string _Name;
        public string Name {
            get { return _Name; }
            set { SetPropertyValue("Name", ref _Name, value); }
        }
    }
}
